tests/unit/FirebaseObject.spec.js

```diff
@@ -322,6 +322,21 @@ describe('$FirebaseObject', function() {
       expect($scope.test).toEqual({$value: null, $id: obj.$id, $priority: obj.$priority});
     });
 
+    it('should delete $value if set to an object', function () {
```

Comment:
Oops sorry. Didn't see your reply. 
Github needs to use Firebase for the line comments system. ;)